hcn/hcnsupport.go

```diff
@@ -43,7 +50,59 @@ type ApiSupport struct {
 	V2 bool `json:"V2"`
 }
 
-// GetSupportedFeatures returns the features supported by the Service.
+// GetCachedSupportedFeatures returns the features supported by the Service and an error if the query failed. If this has been called
+// before it will return the supported features and error received from the first call. This can be used to optimize if many calls to the
+// various hcn.IsXSupported methods need to be made.
+func GetCachedSupportedFeatures() (SupportedFeatures, error) {
+	// Only query the HCN version and features supported once, instead of everytime this is invoked. The logs are useful to
+	// debug incidents where there's confusion on if a feature is supported on the host machine. The sync.Once helps to avoid redundant
+	// spam of these anytime a check needs to be made for if an HCN feature is supported. This is a common occurrence in kube-proxy
+	// for example.
+	featuresOnce.Do(func() {
+		globals, err := GetGlobals()
+		if err != nil {
+			// It's expected if this fails once, it should always fail. It should fail on pre 1803 builds for example.
+			versionErr = errors.Wrap(err, "failed to query HCN version number: this is expected on pre 1803 builds.")
```

Comment:
`versionErr` is named assuming the only error we would need to cache is due to the wrong OS version. I think we should change the variable to be named to reflect its scope (it's the cached error from the supported features check).